Shokofin/Utils/GuardedMemoryCache.cs

```diff
@@ -18,7 +17,9 @@ sealed class GuardedMemoryCache : IDisposable, IMemoryCache
 
     private IMemoryCache Cache;
 
-    private readonly ConcurrentDictionary<object, SemaphoreSlim> Semaphores = new();
+    private static AsyncKeyedLockOptions AsyncKeyedLockOptions = new() { PoolSize = 20, PoolInitialFill = 1 };
```

Comment:
What happens if you have a thousand locks in play? (Which can be possible due to the sheer size of some series and due to the concurrent handling of multiple series/episodes/files at once)?

Reply:
Not a problem. The pool size is not a concurrency limit by any means. Worst case, a new semaphoreslim is created and if the pool is full it doesn't keep adding to it, therefore it adapts depending on load.